src/game/default/g_weapon.c

```diff
@@ -491,6 +509,14 @@ void G_ClientHookThink(g_entity_t *ent) {
 	}
 
 	// send off to the proper sub-function
+
+	if (refire) {
+
+		G_ClientHookCheckFire(ent, true);
+
+		return;
```

Comment:
So this is busted?

Reply:
Probably wrong fire direction.

Reply:
It seems correct, the Z axis is sometimes zero'd though, and I think other axes also got broken sometimes too.